manifests/timer_wrapper.pp

```diff
@@ -0,0 +1,92 @@
+# @summary
+#   Helper to define timer and accompanying services for a given task (cron like interface).
+# @param ensure whether the timer and service should be present or absent
+# @param command the command for the systemd servie to execute
+# @param user the user to run the command as
+# @param on_active_sec run service relative to the time when the timer was activated
+# @param on_boot_sec run service relative to when the machine was booted
+# @param on_start_up_sec run service relative to when the service manager was started
+# @param on_unit_active_sec run service relative to when the unit was last activated
+# @param on_unit_inactive_sec run service relative to when the unit was last deactivated
+# @param on_calendar the calendar event expressions time to run the service
+# @param service_overrides override for the`[Service]` section of the service
+#   example: `'service_overrides' => { 'Wants' => 'network-online.target' }`
+# @param timer_overrides override for the`[Timer]` section of the timer
+#   example: `'timer_overrides' => { 'OnBootSec' => '10' }`
+# @param service_unit_overrides override for the`[Unit]` section of the service
+#   example: `'service_unit_overrides' => { 'Wants' => 'network-online.target' }`
+# @param timer_unit_overrides override for the `[Unit]` section of the timer
+#   can not be used in combination with `additional_timer_params`
+define systemd::timer_wrapper (
+  Enum['present', 'absent']                 $ensure,
+  Optional[String[1]]                       $command = undef,
```

Comment:
As you check below `$command` has to be set. 

Can't you just make it non-optional here in the first place. Also there is an existing type
`Systemd::Unit::Service::Exec` which presumably makes sense here.

Could also default this to `$name`  so e.g

```puppet
systemd::timer_wrapper('/usr/bin/rm -rf /':
   on_boot_sec => '1h',
 }
```
could be done.

Reply:
I made the optional so you don't need to specify it when `ensure = 'absent'` the check is in a dedicated if clause because of this.

Using `Systemd::Unit::Service::Exec` is a good idea ... changed.

I am not a fan of putting the command in the title. This can lead to really ugly resource names in the catalog, also it bad for readability IMHO